### DISCUSSION TITLE
Is there a way to get a preloaded query argument from session/token ?

### DISCUSSION THREAD

**Discussion by MoOx** [RevereCRE/relay-nextjs#65] [Q&A]
I am currently implementing a simple page `/account/edit` and just setup `relay-nextjs` to be able to preload user account values into my form.
I have a simple graphqlquery for this, not a problem here.

The thing is: `relay-nextjs` mentions in the [doc about page api](https://reverecre.github.io/relay-nextjs/docs/page-api#withrelay) that nextjs _route params are taken automatically_ into account (which is very convenient).
**But in my case, I don't have this argument in my route** (and don't really want to have an url with a uuid).

_**I would like to be able to get it from my session**_ (I am using next-auth, working fine, so I have this uuid (as well as my jwt token for queries).

**My question is:** what is the proper place in all this wonderful `relay-nextjs` glue to pass a given argument for a preloaded query ?

I hope my problem is clear, let me know if not !

Have a great day !

**Comment by MoOx** [comment]
It seems [`variablesFromContext`](https://reverecre.github.io/relay-nextjs/docs/page-api#properties) would be what I am looking for but my session is not available in the context.

**Reply by rrdelaney** [reply]
If the auth library allows you to get authentication from the context then it may be able to work! What I've done in this case is create a `viewer` field in my GraphQL API that is resolved on the server fully knowing what auth there is, rather than trying to get who the user is and provide it as a variable.

**Comment by MoOx** [comment]
Solved this problematic by reviewing everything from the ground up with a simple approach (based on https://github.com/vercel/next.js/pull/33892)
